### PR TITLE
HOCS-5561: Use Helm for Deployments

### PR DESCRIPTION
**Pull request by nickmcmahon01**

### CODE REVIEW

File: helm/hocs-management-ui/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v2
+name: hocs-frontend
```
Comment (robertdeniszczyc2) [inline helm/hocs-management-ui/Chart.yaml]:
```suggestion
name: hocs-management-ui
```

File: helm/hocs-management-ui/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+version: 1.0.0
+dependencies:
+  - name: hocs-generic-service
+    version: 1.0.0
+    repository: file:///Users/nicholasmcmahon/Desktop/hocs-helm-charts/charts/generic-service
```
Comment (robertdeniszczyc2) [inline helm/hocs-management-ui/Chart.yaml]: Path is pointing at a local directory

File: helm/hocs-management-ui/templates/_envs.tpl [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+- name: DOCUMENT_SERVICE
+  value: '{{ tpl .Values.app.env.docsService . }}'
+- name: INFO_SERVICE
+  value: '{{ tpl .Values.app.env.infoService . }}'
+- name: OUTBOUND_PROXY
+  value: '{{ tpl .Values.app.env.outboundProxy . }}'
+- name: ENCRYPTION_KEY
+  valueFrom:
+    secretKeyRef:
+      name: hocs-frontend
```
Comment (robertdeniszczyc2) [inline helm/hocs-management-ui/templates/_envs.tpl:71]: This module is hocs-management-ui

File: helm/hocs-management-ui/values.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+---
+hocs-generic-service:
+  nameOverride: hocs-management-ui
+
+  deployment:
+    selectorRole: hocs-frontend
```
Comment (robertdeniszczyc2) [inline helm/hocs-management-ui/values.yaml:6]: This module is hocs-management-ui

**Review by robertdeniszczyc2 — CHANGES_REQUESTED** [review]

File: helm/hocs-management-ui/templates/_envs.tpl (hunk 3 above)
Comment (nickmcmahon01) [reply helm/hocs-management-ui/templates/_envs.tpl:71]: this just a very poorly named secret, 
https://github.com/UKHomeOffice/hocs-management-ui/blob/main/kube/kd/deployment.yaml#L118

File: helm/hocs-management-ui/values.yaml (hunk 4 above)
Comment (nickmcmahon01) [reply helm/hocs-management-ui/values.yaml:6]: this is correct, just confusingly named. it's the 'hocs frontend' network policy not 'hocs-frontend' 
https://github.com/UKHomeOffice/hocs-network-policies/blob/main/helm/hocs-network-policies/templates/hocs-frontend-networkpolicy.yaml

File: helm/hocs-management-ui/Chart.yaml (hunk 2 above)
Comment (nickmcmahon01) [reply helm/hocs-management-ui/Chart.yaml]: good spot, I do this about half the time 👍

**Review by robertdeniszczyc2 — APPROVED** [review]

**Pull request merged**